### PR TITLE
Add Vue.js support

### PR DESCRIPTION
**Pull request by yymm**
Add  a feature for enabling completion on [Single File Components](https://vuejs.org/v2/guide/single-file-components.html) with .vue extension.

Thanks!

### CODE REVIEW

File: plugin/nvim_typescript.vim [hunk 1]
```diff
@@ -77,9 +82,12 @@ augroup nvim-typescript "{{{
   if get(g:, 'nvim_typescript#javascript_support', 1)
     autocmd BufEnter *.ts,*.tsx,*.js,*.jsx call TSOnBufEnter()
```
Comment (Shougo) [inline plugin/nvim_typescript.vim]: It should be:

```vim
autocmd BufEnter *.ts,*.tsx,*.js,*.jsx call TSOnBufEnter()
```

File: plugin/nvim_typescript.vim [hunk 2]
```diff
@@ -77,9 +82,12 @@ augroup nvim-typescript "{{{
...
     autocmd BufWritePost *.ts,*.tsx,*.js,*.jsx call TSOnBufSave()
```
Comment (Shougo) [inline plugin/nvim_typescript.vim]: It should be:

```vim
autocmd BufWritePost *.js,*.jsx call TSOnBufSave()
```

**Comment by yymm** [comment]
@Shougo Thank you for your review, fix it 3b8fc6e .

**Comment by mhartington** [comment]
Thanks!

**Pull request merged**